src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonIncentiveLevelAuditedService.kt

```diff
@@ -38,6 +38,13 @@ class PrisonIncentiveLevelAuditedService(
       }
   }
 
+  override suspend fun deactivateAllPrisonIncentiveLevels(prisonId: String): List<PrisonIncentiveLevelDTO> {
+    return super.deactivateAllPrisonIncentiveLevels(prisonId)
+      .onEach {
+        auditAndPublishPrisonLevelChangeEvent(it)
+      }
```

Comment:
As the `deactivateAllPrisonIncentiveLevels()` currently returns all levels for a prison - even the one which were already inactive - this would publish 'changed' domain events for prison levels that didn't really change.

Reply:
i figured that a few superfluous change events would not a big deal. given that the api’s caller wants the whole set to be inactive, it’s not bad to pretend that they all changed even if some did not. anyway, it’s addressed now :)